shell/egl_window.h

```diff
@@ -27,7 +27,13 @@
 #include "egl.h"
 #include "xdg-shell-client-protocol.h"
 
+#define WL_ARRAY_FOR_EACH(pos, array, type)                             \
```

Comment:
#ifndef ....

Please also explain why we need this.... I'd like to see some cpp construct that fixes this... seems a bit weird that can't really cast generic pointers. In C we don't even need an explicit cast.

Reply:
The base Wayland macro does not compile as-is in C++.  This is pretty well documented on the internet.  I added a comment in updated patches.

Reply:
https://gitlab.freedesktop.org/wayland/wayland/-/issues/34